new-backend/app/db/vector_store.py

```
@@ -0,0 +1,41 @@
+
+# Load Pinecone credentials from environment variables
+PINECONE_API_KEY = os.getenv("PINECONE_API_KEY")
+if not PINECONE_API_KEY:
+    raise ValueError("PINECONE_API_KEY environment variable is required")
+try:
+
+    # Initialize Pinecone client
+    pc = Pinecone(api_key=PINECONE_API_KEY)
+
+except Exception as e:
+    raise RuntimeError(f"Error occured while intialising pinecone client:{e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)